src/docs/deployment/obfuscate.md

```diff
@@ -83,9 +83,24 @@ flutter symbolize -i <stack trace file> -d /out/android/app.android-arm64.symbol
    For more information on the `symbolize` command,
    run `flutter symbolize -h`.
 
+## Caveat
+
+Be aware of the following when coding an app that will
+eventually be an obfuscated binary.
+
+* Code that relies on matching specific class, function,
+  or library names will fail.
+  For example, the following call to `expect()` will not
+  work in an obfuscated binary:
+
+```dart
+expect(Foo.runtimeType.toString(), equals('Foo'))
```

Comment:
I would make it more obvious this is an instance instead of the class, since Foo.runtimeType should be `Type`

```suggestion
expect(foo.runtimeType.toString(), equals('Foo'))
```